main/ejml-core/test/org/ejml/data/TestDMatrixSparseCSC.java

```diff
@@ -45,25 +45,30 @@ public boolean isStructureValid(DMatrixSparse m) {
         return CommonOps_DSCC.checkStructure((DMatrixSparseCSC)m);
     }
 
+    @Test
+    public void constructor_tooLarge() {
+        assertThrows(IllegalArgumentException.class,()->new DMatrixSparseCSC(1_000_000_000,1_000_000_000,4));
+    }
+
     @Test
     public void constructor_veryLarge() {
-        DMatrixSparseCSC a = new DMatrixSparseCSC(1_000_000_000,100_000_000,4);
+        DMatrixSparseCSC a = new DMatrixSparseCSC(1_000_000,1_000,4);
```

Comment:
I think this behaviour should still be allowed, as its not wrong to my point of view

Reply:
`nz_rows` can only take on values from 0 to `Integer.MAX_VALUE` if `numRows*numCols` exceeds that value then not every element can be referenced. The long term fix is to change it from int to long, pun intended.